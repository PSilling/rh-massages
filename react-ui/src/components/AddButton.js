// react imports
import React, { Component } from 'react';

/**
 * Custom add button component.
 */
class AddButton extends Component {

  render() {
    return(
      <span>
        <button type="button" className="btn" onClick={this.props.onAdd}>
          <span className="glyphicon glyphicon-plus"></span>
        </button>
      </span>
    );
  }
}

export default AddButton
