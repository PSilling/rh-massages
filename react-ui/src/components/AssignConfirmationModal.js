// react imports
import React, { Component } from 'react';

// component imports
import EventModalActions from './EventModalActions';

// module imports
import {ModalContainer, ModalDialog} from 'react-modal-dialog';

// util imports
import _t from '../utils/Translations.js';

/**
 * Custom confirmation modal component.
 */
class AssignConfirmationModal extends Component {

  render() {
    return(
      <ModalContainer onClose={this.props.onClose}>
        <ModalDialog onClose={this.props.onClose} width="40%">
            <h3>{ _t.translate('Action confirmation') }</h3>
            <p>{this.props.message}</p>
            <EventModalActions
              primaryWithEventLabel={ _t.translate('Proceed and add to calendar') }
              onProceedWithEvent={this.props.onConfirmWithEvent}
              primaryLabel={ _t.translate('Proceed') }
              onProceed={this.props.onConfirm}
              onClose={this.props.onClose}
            />
          </ModalDialog>
      </ModalContainer>
    )
  }
}

export default AssignConfirmationModal
