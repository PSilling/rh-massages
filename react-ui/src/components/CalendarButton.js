// react imports
import React, { Component } from 'react';

/**
 * Custom button component that handles event addition to Google Calendar.
 */
class CalendarButton extends Component {

  render() {
    return(
      <span>
        <button disabled={this.props.disabled} style={{ 'color': '#000' }}  type="button"
          className="btn btn-link" onClick={this.props.onAdd}>
          <span className="glyphicon glyphicon-calendar"></span>
        </button>
      </span>
    );
  }
}

export default CalendarButton
