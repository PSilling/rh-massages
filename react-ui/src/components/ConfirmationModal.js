// react imports
import React, { Component } from 'react';

// component imports
import ModalActions from './ModalActions';

// module imports
import {ModalContainer, ModalDialog} from 'react-modal-dialog';

// util imports
import _t from '../utils/Translations.js';

/**
 * Custom confirmation modal component.
 */
class ConfirmationModal extends Component {

  render() {
    return(
      <ModalContainer onClose={this.props.onClose}>
        <ModalDialog onClose={this.props.onClose} width="40%">
            <h3>{ _t.translate('Action confirmation') }</h3>
            <p>{this.props.message}</p>
            <ModalActions
              primaryLabel={ _t.translate('Proceed') }
              onProceed={this.props.onConfirm}
              onClose={this.props.onClose}
            />
          </ModalDialog>
      </ModalContainer>
    )
  }
}

export default ConfirmationModal
