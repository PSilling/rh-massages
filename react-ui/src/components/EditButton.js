// react imports
import React, { Component } from 'react';

/**
 * Custom edit button component.
 */
class EditButton extends Component {

  render() {
    return(
      <span>
        <button style={{ 'color': '#000' }}  type="button" className="btn btn-link" onClick={this.props.onEdit}>
          <span className="glyphicon glyphicon-pencil"></span>
        </button>
      </span>
    );
  }
}

export default EditButton
