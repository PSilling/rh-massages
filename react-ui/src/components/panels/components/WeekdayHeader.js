// react imports
import React, { Component } from "react";
import PropTypes from "prop-types";

/**
 * Calendar day header for week view.
 */
class Header extends Component {
  handleKeyPress = event => {
    if (event.key === "Space") {
      this.props.onClick(this.props.date);
    }
  };

  render() {
    if (this.props.active) {
      return (
        <span
          role="button"
          tabIndex="-3"
          style={{ border: "none", outline: "none" }}
          onClick={() => this.props.onClick(this.props.date)}
          onKeyPress={this.handleKeyPress}
        >
          {this.props.label}
        </span>
      );
    }

    return <span style={{ cursor: "default" }}>{this.props.label}</span>;
  }
}

const WeekdayHeader = (props, active, onClick) => <Header {...props} active={active} onClick={onClick} />;

Header.propTypes = {
  date: PropTypes.instanceOf(Date).isRequired,
  label: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  active: PropTypes.bool
};

Header.defaultProps = {
  active: false
};

export default WeekdayHeader;
