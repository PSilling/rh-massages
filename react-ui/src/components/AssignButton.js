// react imports
import React, { Component } from 'react';

// component imports
import ConfirmationModal from '../components/ConfirmationModal';

// util imports
import _t from '../utils/Translations.js';

/**
 * Custom assign myself button component.
 */
class AssignButton extends Component {

  state = {active: false}

  handleToggle = () => {
    this.setState({active: !this.state.active});
  }

  render() {
    return(
      <span>
        <button type="button" className="btn btn-success pull-right" onClick={this.handleToggle}>
          { _t.translate('Assign me') }
        </button>
        {this.state.active ?
          <ConfirmationModal
            message={ _t.translate('Are you sure you want to assign yourself to this massage?') }
            onClose={this.handleToggle}
            onConfirm={() => {
              this.handleToggle();
              this.props.onAssign();
            }}
          /> : ''
        }
      </span>
    )
  }
}

export default AssignButton
