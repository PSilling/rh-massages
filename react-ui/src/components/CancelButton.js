// react imports
import React, { Component } from 'react';

// component imports
import ConfirmationModal from '../components/ConfirmationModal';

// util imports
import _t from '../utils/Translations.js';

/**
 * Custom cancel button component.
 */
class CancelButton extends Component {

  state = {active: false}

  handleToggle = () => {
    this.setState({active: !this.state.active});
  }

  render() {
    return(
      <span>
        <button type="button" className="btn btn-warning pull-right" onClick={this.handleToggle}>
          { _t.translate('Cancel') }
        </button>
        {this.state.active ?
          <ConfirmationModal
            message={ _t.translate('Are you sure you want to cancel this massage?') }
            onClose={this.handleToggle}
            onConfirm={() => {
              this.handleToggle();
              this.props.onCancel();
            }}
          /> : ''
        }
      </span>
    )
  }
}

export default CancelButton
