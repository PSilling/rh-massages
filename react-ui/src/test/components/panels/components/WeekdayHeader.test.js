// react imports
import React from "react";
import { shallow } from "enzyme";

// test imports
import { Header } from "../../../../components/panels/components/WeekdayHeader";

test("renders content with correct props", () => {
  const testFunction = jest.fn();
  const testDate = new Date(0);
  const wrapper = shallow(<Header active label="testHeader" date={testDate} onClick={testFunction} />);
  const spans = wrapper.find("span");

  expect(spans.length).toBe(1);
  expect(spans.props().children).toEqual("testHeader");
  spans.props().onClick();
  expect(testFunction).toHaveBeenCalledTimes(1);
  expect(testFunction).toHaveBeenLastCalledWith(testDate);

  expect(wrapper).toMatchSnapshot();
});
