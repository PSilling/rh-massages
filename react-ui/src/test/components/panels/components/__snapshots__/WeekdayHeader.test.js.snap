// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders content with correct props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    active={true}
    date={1970-01-01T00:00:00.000Z}
    label="testHeader"
    onClick={
      [MockFunction] {
        "calls": Array [
          Array [
            1970-01-01T00:00:00.000Z,
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "testHeader",
      "onClick": [Function],
      "onKeyPress": [Function],
      "role": "button",
      "style": Object {
        "border": "none",
        "outline": "none",
      },
      "tabIndex": "-3",
    },
    "ref": null,
    "rendered": "testHeader",
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "testHeader",
        "onClick": [Function],
        "onKeyPress": [Function],
        "role": "button",
        "style": Object {
          "border": "none",
          "outline": "none",
        },
        "tabIndex": "-3",
      },
      "ref": null,
      "rendered": "testHeader",
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
